### PR TITLE
Filter optimization with range index in fwd dictionary cache lookup

### PR DESCRIPTION
**Pull request by sujith71955**
Below implementation is been done for providing optimization in forward dictionary cache look up while applying filters, a range based index is provided for faster look up so that always the dictionary look up will start based on provided range

### CODE REVIEW

**Comment by sujith71955** [comment]
retest this please

**Comment by sujith71955** [comment]
retest this please

File: core/src/main/java/org/carbondata/core/carbon/datastore/SegmentTaskIndexStore.java [hunk 1]
```diff
@@ -83,6 +85,12 @@ public static SegmentTaskIndexStore getInstance() {
     return SEGMENTTASKINDEXSTORE;
   }
 
+  public static void main(String[] args) {
+    Path p = new Path("file:/D:/file");
+    Path pathWithoutSchemeAndAuthority = Path.getPathWithoutSchemeAndAuthority(p);
```
Comment (gvramana) [inline core/src/main/java/org/carbondata/core/carbon/datastore/SegmentTaskIndexStore.java]: remove this

File: core/src/main/java/org/carbondata/query/filters/measurefilter/util/FilterUtil.java [hunk 2]
```diff
@@ -244,14 +246,11 @@ public static DimColumnFilterInfo getFilterValues(AbsoluteTableIdentifier tableI
...
-      // Reading the dictionary value from cache.
-      Integer dictionaryVal =
-          getDictionaryValue(resultVal, tableIdentifier, columnExpression.getDimension());
-      if (null != dictionaryVal) {
-        surrogates.add(dictionaryVal);
-      }
-    }
+    Stack<Integer> startIndexContainer = new Stack<Integer>();
+    startIndexContainer.push(0);
+    // Reading the dictionary value from cache.
```
Comment (gvramana) [inline core/src/main/java/org/carbondata/query/filters/measurefilter/util/FilterUtil.java]: remove this stack now

**Comment by gvramana** [comment]
closing as re-raised as #570

**Pull request closed without merging**